jetty-xml/src/test/java/org/eclipse/jetty/xml/XmlConfigurationTest.java

```diff
@@ -1309,6 +1318,185 @@ public void testJettyStandardIdsAndPropertiesAndJettyHomeUriAndJettyBaseUri() th
+                "</Configure>");
+
+        IllegalStateException cause = assertThrows(IllegalStateException.class, () ->
+            mimicXmlConfigurationMain(xmlBar));
+
+        assertThat("Cause message", cause.getMessage(), containsString("No matching constructor"));
+    }
+
+    @Test
+    public void testConfiguredSameWithNamedArgTwice() throws Exception
+    {
+        XmlConfiguration xmlFoo = asXmlConfiguration("foo.xml",
+            "<Configure>\n" +
+                "  <New id=\"foo\" class=\"java.lang.String\">\n" +
+                "    <Arg>foozball</Arg>\n" +
+                "  </New>\n" +
+                "</Configure>");
+        XmlConfiguration xmlBar = asXmlConfiguration("bar.xml",
+            "<Configure id=\"bar\" class=\"" + BarNamed.class.getName() + "\">\n" +
+                "  <Arg name=\"foo\"><Ref refid=\"foo\"/></Arg>\n" +
+                "</Configure>");
+        XmlConfiguration xmlAddZed = asXmlConfiguration("zed.xml",
+            "<Configure id=\"bar\" class=\"" + BarNamed.class.getName() + "\">\n" +
+                "  <Arg name=\"foo\">baz</Arg>\n" + // the invalid line
+                "  <Call name=\"addZed\">\n" +
+                "    <Arg>plain-zero</Arg>\n" +
+                "  </Call>\n" +
+                "</Configure>");
+
+        try (StdErrCapture logCapture = new StdErrCapture(XmlConfiguration.class))
+        {
+            Map<String, Object> idMap = mimicXmlConfigurationMain(xmlFoo, xmlBar, xmlAddZed);
+            Object obj = idMap.get("bar");
+            assertThat("BarNamed instance created", obj, instanceOf(BarNamed.class));
+            BarNamed bar = (BarNamed)obj;
+            assertThat("BarNamed has foo", bar.getFoo(), is("foozball"));
+            List<String> zeds = bar.getZeds();
+            assertThat("BarNamed has zeds", zeds, not(empty()));
+            assertThat("Zeds[0]", zeds.get(0), is("plain-zero"));
+
+            List<String> warnLogs = logCapture.getLines()
+                .stream().filter(line -> line.contains(":WARN:"))
+                .collect(Collectors.toList());
+
+            assertThat("WARN logs count", warnLogs.size(), is(1));
+
+            String actualWarn = warnLogs.get(0);
+            assertThat("WARN logs", actualWarn,
+                allOf(containsString("Ignored arg <Arg name="),
+                    containsString("zed.xml")
+                ));
+        }
+    }
+
+    /**
+     * This mimics the XML load behavior in XmlConfiguration.main(String ... args)
+     */
+    private Map<String, Object> mimicXmlConfigurationMain(XmlConfiguration... configurations) throws Exception
+    {
+        XmlConfiguration last = null;
+        for (XmlConfiguration configuration : configurations)
+        {
+            if (last != null)
+                configuration.getIdMap().putAll(last.getIdMap());
+            configuration.configure();
+            last = configuration;
+        }
+        return last.getIdMap();
+    }
```

Comment:
I would like to see a test where there is a `Arg` misplaced, as below, so that we define in a test how we should behave.

```xml
<Configure ...>
  <Arg>...</Arg>
  <New ...>...</New>
  <Arg>...</Arg>
</Configure>
```

Reply:
@gregw pointed out that the Configure_9_3.dtd (and Configure_10_0.dtd) does not allow a `<Arg>` anywhere but the start of the element list.

https://github.com/eclipse/jetty.project/blob/678385bfda0f90a0bcccf7f86aa272073c1b4f29/jetty-xml/src/main/resources/org/eclipse/jetty/xml/configure_10_0.dtd#L45-L46

Reply:
A test that is expected to fail in this scenario is dependent on the dtd used (or not used) I would think.

Reply:
Then shouldn't we do the same for `Call` and `New`? I don't think the DTD has the same enforcement for those.

Reply:
Opened #4658 for these new test cases.

Also `<Call>`

https://github.com/eclipse/jetty.project/blob/678385bfda0f90a0bcccf7f86aa272073c1b4f29/jetty-xml/src/main/resources/org/eclipse/jetty/xml/configure_10_0.dtd#L168-L169

And `<New>`

https://github.com/eclipse/jetty.project/blob/678385bfda0f90a0bcccf7f86aa272073c1b4f29/jetty-xml/src/main/resources/org/eclipse/jetty/xml/configure_10_0.dtd#L209-L210

Reply:
Also opened #4659 for the missing `Id?,Name?,Class?,Arg*` in the `<Configure>` section on `Configure_10_0.dtd`

Reply:
@sbordet we had such a test, but that is invalid XML against the DTD.  I don't think we should be required to process non-valid XML

Note that I'm doing a cleanup in 10 - see #4656